### PR TITLE
docs(adoptium-java datasource): rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Rewrite document

## Context

This document needed some work. I hope this is easier to understand now.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/adoptium-java/readme.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-This datasource returns releases from [Adoptium](https://adoptium.net/) API.
+This datasource returns releases from the [Adoptium](https://adoptium.net/) API.
 
 It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
 
 It only uses the first 50 pages with 50 items per page.
```
Comment (HonkingGoose) [inline lib/modules/datasource/adoptium-java/readme.md:5]: > It only uses the first 50 pages with 50 items per page.

This is confusing.

How about something like this instead?

```suggestion
When Renovate contacts the Adoptium API, it fetches 50 pages.
Each page has 50 items.
So 2500 items are fetched from the API in total.
```

File: lib/modules/datasource/adoptium-java/readme.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
-This datasource returns releases from [Adoptium](https://adoptium.net/) API.
+This datasource returns releases from the [Adoptium](https://adoptium.net/) API.
 
 It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.
```
Comment (HonkingGoose) [inline lib/modules/datasource/adoptium-java/readme.md:3]: I'm not sure what `filter parameters` means here. The `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` is a bit tricky to parse on first glance.

How about we add a plain language explanation like this maybe?

The datasource will check:

- JRE or JDK files
- with a JDK project
- which have the GA (general availability) status
- and sorts the results in descending order
- only checks vendor type Adoptium

**Review by HonkingGoose — COMMENTED** [review]
We can probably clarify the text even more. Let me know what you think. 😉

File: lib/modules/datasource/adoptium-java/readme.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
```
Comment (viceice) [inline lib/modules/datasource/adoptium-java/readme.md:3]:
```suggestion
It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC` as filter parameters.
```

File: lib/modules/datasource/adoptium-java/readme.md (hunk 2 above)
Comment (viceice) [reply lib/modules/datasource/adoptium-java/readme.md:3]: sure, vendor filter has been removed. I Don't really know the details of the `project` filter. I use mainly the default values

https://api.adoptium.net/q/swagger-ui/#/Release%20Info/getReleaseVersions

**Review by rarkins — DISMISSED** [review]

File: lib/modules/datasource/adoptium-java/readme.md [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
-Use `java-jre` as `packageName` if you only want releases which come with the Java Runtime Environment.
-Currently only the LTS releases of Java come with the JRE.
+When Renovate contacts the Adoptium API, it fetches 50 pages.
+Each page has 50 items.
+So 2500 items are fetched from the API in total.
+
+If you want to get releases which come with the JDK, set the `packageName` to `java-jdk` or `java`.
+
+If you want to get releases which come with the JRE, set the `packageName` to `java-jre`.
+Currently, only LTS releases of Java come with the JRE.
```
Comment (viceice) [inline lib/modules/datasource/adoptium-java/readme.md:19]: it seems this is no longer true, i saw are for non lts too recently

Comment (HonkingGoose) [reply lib/modules/datasource/adoptium-java/readme.md:19]:
```suggestion
```

I think we can drop the line then?

Comment (viceice) [reply lib/modules/datasource/adoptium-java/readme.md:19]: maybe we should leave something like the isn't always a jre for each version?

Comment (rarkins) [reply lib/modules/datasource/adoptium-java/readme.md:19]: @viceice I think you're the only on who understands it enough to write it :)

Comment (HonkingGoose) [reply lib/modules/datasource/adoptium-java/readme.md:19]: How about this then?

```suggestion
LTS releases of Java will have a version that comes with the JRE.
Non-LTS release may not always include the JRE.
```

**Review by viceice — DISMISSED** [review]

File: lib/modules/datasource/adoptium-java/readme.md [hunk 5]
```diff
@@ -16,4 +16,5 @@ So 2500 items are fetched from the API in total.
 If you want to get releases which come with the JDK, set the `packageName` to `java-jdk` or `java`.
 
 If you want to get releases which come with the JRE, set the `packageName` to `java-jre`.
-Currently, only LTS releases of Java come with the JRE.
+LTS releases of Java will have a version that comes with the JRE.
+Non-LTS release may not always include the JRE.
```
Comment (HonkingGoose) [inline lib/modules/datasource/adoptium-java/readme.md:20]:
```suggestion
Non-LTS releases may not always include the JRE.
```

Whoops, it should be "releases" here. 🙈

**Review by viceice — APPROVED** [review]

**Pull request merged**